laws/src/main/scala/cats/laws/SelectiveLaws.scala

```diff
@@ -0,0 +1,53 @@
+package cats
+package laws
+
+import cats.syntax.all._
+
+/**
+ * Laws that must be obeyed by any `Selective`.
```

Comment:
if none of these laws require rigidity, why not put them on Applicative while testing selectA?

Reply:
Will address in [this thread](https://github.com/typelevel/cats/pull/3709#discussion_r544741355).